mongo/search_index_prose_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
```

Comment:
Should this file be in the `integration` dir with the other prose tests?

Reply:
The motivation for placing it separately is that this file is run directly from `Makefile` by `go test` rather than those in the `integration` dir.

Reply:
OK, I've moved it to 'mongo\integration`.